build_runner/lib/src/server/server.dart

```diff
@@ -39,8 +44,24 @@ class ServeHandler implements BuildState {
 
   FutureOr<Response> _handle(Request request, String rootDir) async {
     await currentBuild;
+    if (_lastBuildResult.status == BuildStatus.failure) {
+      return new Response(200,
+          body: '<h1>Build failed!</h1>'
```

Comment:
I think it could be worth pulling this out into a separate method, otherwise it takes up more space than it earns in this one.

Reply:
done